## Exercice 1

### Question 1

In [1]:
lettres_accentuees = ['é', 'è', 'ê', 'à', 'â', 'ï', 'ç', 'ù']
for lettre in lettres_accentuees:
    code_numerique = ord(lettre)
    print(f"Le code numérique de {lettre} est {code_numerique}")

Le code numérique de é est 233
Le code numérique de è est 232
Le code numérique de ê est 234
Le code numérique de à est 224
Le code numérique de â est 226
Le code numérique de ï est 239
Le code numérique de ç est 231
Le code numérique de ù est 249


### Question 3

In [3]:
def flemme(code_debut, code_fin):
    for chiffre in range(code_debut, code_fin + 1):
        lettre = chr(chiffre)
        print(f"Code {chiffre} : {lettre}")

flemme(97, 122)  # Affiche les lettres de 'a' à 'z'

Code 97 : a
Code 98 : b
Code 99 : c
Code 100 : d
Code 101 : e
Code 102 : f
Code 103 : g
Code 104 : h
Code 105 : i
Code 106 : j
Code 107 : k
Code 108 : l
Code 109 : m
Code 110 : n
Code 111 : o
Code 112 : p
Code 113 : q
Code 114 : r
Code 115 : s
Code 116 : t
Code 117 : u
Code 118 : v
Code 119 : w
Code 120 : x
Code 121 : y
Code 122 : z


### Question 4

La sequence binaire "01000010011100100110000101110110011011110010000000100001" décodé en texte donne le mot "Bravo !"

### Question 5

## Exercice 2


### Question 1

La séquence qui représente le "é" est: 1110 1001. En UTF-8 on le code sur deux octets en respectant les précisions apportées dans le tableau ci-dessus. Les bits imposés sont 110 pour le 1er octet et 10 pour le deuxième, le code du « é » est écrit en commençant par la droite et l’octet de gauche est rempli par des zéros (en italique). Voilà ce que l’on obtient : 11000011 10101001.

### Question 2

Si le fichier avait été interprété en latin 1